File: dart/lib/model/v3_siri_reference_data_detail.dart
part of swagger.api;

class V3SiriReferenceDataDetail {
  
  int routeId = null;
/* Route number */
  String routeNumberShort = null;

  int directionId = null;
/* Authority (Upstream SIRI provider) of a route and direction */
  int trackingSupplierId = null;

  int routeType = null;

  V3SiriReferenceDataDetail();

  @override
  String toString() {
    return 'V3SiriReferenceDataDetail[routeId=$routeId, routeNumberShort=$routeNumberShort, directionId=$directionId, trackingSupplierId=$trackingSupplierId, routeType=$routeType, ]';
  }

  V3SiriReferenceDataDetail.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    routeId = json['route_id'];
    routeNumberShort = json['route_number_short'];
    directionId = json['direction_id'];
    trackingSupplierId = json['tracking_supplier_id'];
    routeType = json['route_type'];
  }

  Map<String, dynamic> toJson() {
    return {
      'route_id': routeId,
      'route_number_short': routeNumberShort,
      'direction_id': directionId,
      'tracking_supplier_id': trackingSupplierId,
      'route_type': routeType
     };
  }

  static List<V3SiriReferenceDataDetail> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriReferenceDataDetail>() : json.map((value) => new V3SiriReferenceDataDetail.fromJson(value)).toList();
  }

  static Map<String, V3SiriReferenceDataDetail> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriReferenceDataDetail>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriReferenceDataDetail.fromJson(value));
    }
    return map;
  }
}

File: java-vertx/src/main/java/io/swagger/server/api/model/V3SiriReferenceDataDetail.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3SiriReferenceDataDetail
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3SiriReferenceDataDetail   {
  private Integer routeId = null;

  private String routeNumberShort = null;

  private Integer directionId = null;

  private Integer trackingSupplierId = null;

  private Integer routeType = null;

  public V3SiriReferenceDataDetail routeId(Integer routeId) {
    this.routeId = routeId;
    return this;
  }

  /**
   * Get routeId
   * @return routeId
   **/
    public Integer getRouteId() {
    return routeId;
  }

  public void setRouteId(Integer routeId) {
    this.routeId = routeId;
  }

  public V3SiriReferenceDataDetail routeNumberShort(String routeNumberShort) {
    this.routeNumberShort = routeNumberShort;
    return this;
  }

  /**
   * Route number
   * @return routeNumberShort
   **/
    public String getRouteNumberShort() {
    return routeNumberShort;
  }

  public void setRouteNumberShort(String routeNumberShort) {
    this.routeNumberShort = routeNumberShort;
  }

  public V3SiriReferenceDataDetail directionId(Integer directionId) {
    this.directionId = directionId;
    return this;
  }

  /**
   * Get directionId
   * @return directionId
   **/
    public Integer getDirectionId() {
    return directionId;
  }

  public void setDirectionId(Integer directionId) {
    this.directionId = directionId;
  }

  public V3SiriReferenceDataDetail trackingSupplierId(Integer trackingSupplierId) {
    this.trackingSupplierId = trackingSupplierId;
    return this;
  }

  /**
   * Authority (Upstream SIRI provider) of a route and direction
   * @return trackingSupplierId
   **/
    public Integer getTrackingSupplierId() {
    return trackingSupplierId;
  }

  public void setTrackingSupplierId(Integer trackingSupplierId) {
    this.trackingSupplierId = trackingSupplierId;
  }

  public V3SiriReferenceDataDetail routeType(Integer routeType) {
    this.routeType = routeType;
    return this;
  }

  /**
   * Get routeType
   * @return routeType
   **/
    public Integer getRouteType() {
    return routeType;
  }

  public void setRouteType(Integer routeType) {
    this.routeType = routeType;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3SiriReferenceDataDetail v3SiriReferenceDataDetail = (V3SiriReferenceDataDetail) o;
    return Objects.equals(this.routeId, v3SiriReferenceDataDetail.routeId) &&
        Objects.equals(this.routeNumberShort, v3SiriReferenceDataDetail.routeNumberShort) &&
        Objects.equals(this.directionId, v3SiriReferenceDataDetail.directionId) &&
        Objects.equals(this.trackingSupplierId, v3SiriReferenceDataDetail.trackingSupplierId) &&
        Objects.equals(this.routeType, v3SiriReferenceDataDetail.routeType);
  }

  @Override
  public int hashCode() {
    return Objects.hash(routeId, routeNumberShort, directionId, trackingSupplierId, routeType);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3SiriReferenceDataDetail {\n");
    
    sb.append("    routeId: ").append(toIndentedString(routeId)).append("\n");
    sb.append("    routeNumberShort: ").append(toIndentedString(routeNumberShort)).append("\n");
    sb.append("    directionId: ").append(toIndentedString(directionId)).append("\n");
    sb.append("    trackingSupplierId: ").append(toIndentedString(trackingSupplierId)).append("\n");
    sb.append("    routeType: ").append(toIndentedString(routeType)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
